File: Bwl.Imaging/Geometry/Parameters/Parameter.vb
Public Class Parameter
    Public Property Key As String = String.Empty
    Public Property Value As String = String.Empty
    Public Property Unit As String = String.Empty
    Public Property Visible As Boolean = True
    Public Property Caption As String = String.Empty
    Public Property AdditionalSettings As String = String.Empty

    Public Sub New()
    End Sub

    Public Sub New(key As String, value As String)
        Me.Key = key
        Me.Value = value
    End Sub

    Public Sub New(key As String, value As String, visible As Boolean)
        Me.Key = key
        Me.Value = value
        Me.Visible = visible
    End Sub

    Public Sub New(key As String, value As String, unit As String, visible As Boolean, caption As String, additionalSettings As String)
        Me.Key = key
        Me.Value = value
        Me.Unit = unit
        Me.Visible = visible
        Me.Caption = caption
        Me.AdditionalSettings = additionalSettings
    End Sub
End Class

File: Bwl.Imaging.Skia/Geometry/Parameters/Parameter.cs
namespace Bwl.Imaging;

public class Parameter
{
    public string Key { get; set; } = string.Empty;
    public string Value { get; set; } = string.Empty;
    public string Unit { get; set; } = string.Empty;
    public bool Visible { get; set; } = true;
    public string Caption { get; set; } = string.Empty;
    public string AdditionalSettings { get; set; } = string.Empty;

    public Parameter()
    {
    }

    public Parameter(string key, string value)
    {
        Key = key;
        Value = value;
    }

    public Parameter(string key, string value, bool visible)
    {
        Key = key;
        Value = value;
        Visible = visible;
    }

    public Parameter(string key, string value, string unit, bool visible, string caption, string additionalSettings)
    {
        Key = key;
        Value = value;
        Unit = unit;
        Visible = visible;
        Caption = caption;
        AdditionalSettings = additionalSettings;
    }
}